apps/api/src/app/workflows-v2/usecases/generate-preview/generate-preview.usecase.ts

```diff
@@ -45,24 +55,80 @@ export class GeneratePreviewUsecase {
     };
   }
 
-  private async getValidatedContent(dto: GeneratePreviewRequestDto, stepData: StepDataDto, user: UserSessionData) {
+  private async findWorkflow(command: GeneratePreviewCommand) {
+    return await this.getWorkflowByIdsUseCase.execute(
+      GetWorkflowByIdsCommand.create({
+        identifierOrInternalId: command.identifierOrInternalId,
+        environmentId: command.user.environmentId,
+        organizationId: command.user.organizationId,
+        userId: command.user._id,
+      })
+    );
+  }
+
+  /**
+   * Generates a payload based solely on the schema.
+   * Supports nested schemas and applies defaults where defined.
+   * @param JSONSchemaDto - Defining the structure. example:
+   *  {
+   *    firstName: { type: 'string', default: 'John' },
+   *    lastName: { type: 'string' }
+   *  }
+   * @returns - Generated payload. example: { firstName: 'John', lastName: '{{payload.lastName}}' }
+   */
+  private generatePayloadVariableExample(schema: JSONSchemaDto, path = 'payload'): Record<string, unknown> {
+    if (schema.type !== 'object' || !schema.properties) {
+      throw new Error('Schema must define an object with properties.');
+    }
+
+    return Object.entries(schema.properties).reduce((acc, [key, definition]) => {
+      if (typeof definition === 'boolean') {
+        return acc;
+      }
+
+      const currentPath = `${path}.${key}`;
+
+      if (definition.default) {
+        acc[key] = definition.default;
+      } else if (definition.type === 'object' && definition.properties) {
+        acc[key] = this.generatePayloadVariableExample(definition, currentPath);
+      } else {
+        acc[key] = `{{${currentPath}}}`;
+      }
+
+      return acc;
+    }, {});
+  }
+
+  private async getValidatedContent(
+    dto: GeneratePreviewRequestDto,
+    stepData: StepDataDto,
+    user: UserSessionData,
+    payloadSchema: JSONSchemaDto
+  ) {
     if (!stepData.controls?.dataSchema) {
       throw new StepMissingControlsException(stepData.stepId, stepData);
     }
 
+    const payloadVariablesFromSchema = payloadSchema ? this.generatePayloadVariableExample(payloadSchema) : {};
+    const payloadVariablesExample = { ...payloadVariablesFromSchema, ...dto.previewPayload?.payload };
```

Comment:
i'm not sure this does deep merge, we need to be curfull here, we should have a deep merging here, also